### PR TITLE
AWS S3: adapt to Akka 2.6

### PR DESCRIPTION
**Pull request by ennru**
The bucket management API with `Materializer` is now deprecated.

References #2494 and #2443

### CODE REVIEW

File: s3/src/main/scala/akka/stream/alpakka/s3/javadsl/S3.scala [hunk 1]
```diff
@@ -820,6 +862,47 @@ object S3 {
...
+  /**
+   * Delete bucket with a given name
+   *
+   * @see https://docs.aws.amazon.com/AmazonS3/latest/API/API_DeleteBucket.html
+   * @param bucketName   bucket name
+   * @param system the actor system which provides the materializer to run with
+   * @param attributes attributes to run request with
+   * @return [[java.util.concurrent.CompletionStage CompletionStage]] of type [[Done]] as API doesn't return any additional information
+   */
+  def deleteBucket(bucketName: String,
```
Comment (seglo) [inline s3/src/main/scala/akka/stream/alpakka/s3/javadsl/S3.scala:874]: Should we deprecate the other overloads that take a `Materializer`?

**Comment by ennru** [comment]
Travis does actually build this PR in https://travis-ci.com/github/akka/alpakka/builds/197176252

File: s3/src/test/scala/akka/stream/alpakka/s3/impl/MemoryBufferSpec.scala [hunk 2]
```diff
@@ -26,16 +26,17 @@ class MemoryBufferSpec(_system: ActorSystem)
...
 
-  implicit val materializer = ActorMaterializer(ActorMaterializerSettings(system).withDebugLogging(true))
+  private val DebugLogging = ActorAttributes.debugLogging(true)
 
   override protected def afterAll(): Unit = TestKit.shutdownActorSystem(system)
 
   "MemoryBuffer" should "emit a chunk on its output containg the concatenation of all input values" in {
     val result = Source(Vector(ByteString(1, 2, 3, 4, 5), ByteString(6, 7, 8, 9, 10, 11, 12), ByteString(13, 14)))
       .via(new MemoryBuffer(200))
+      .withAttributes(DebugLogging)
```
Comment (seglo) [inline s3/src/test/scala/akka/stream/alpakka/s3/impl/MemoryBufferSpec.scala:39]: Do we want to keep this?

File: s3/src/test/scala/akka/stream/alpakka/s3/impl/S3StreamSpec.scala [hunk 3]
```diff
@@ -160,7 +159,8 @@ class S3StreamSpec(_system: ActorSystem)
     def bucketStatusPreparation(response: HttpResponse): Future[BucketAccess] = {
       val testedMethod = PrivateMethod[Future[BucketAccess]]('processCheckIfExistsResponse)
```
Comment (seglo) [inline s3/src/test/scala/akka/stream/alpakka/s3/impl/S3StreamSpec.scala:160]:
```suggestion
      val testedMethod = PrivateMethod[Future[BucketAccess]](Symbol("processCheckIfExistsResponse"))
```

File: s3/src/test/scala/akka/stream/alpakka/s3/impl/S3StreamSpec.scala [hunk 4]
```diff
@@ -160,7 +159,8 @@ class S3StreamSpec(_system: ActorSystem)
...
 
-      val result: Future[BucketAccess] = S3Stream invokePrivate testedMethod(response, materializer)
+      val result: Future[BucketAccess] = S3Stream invokePrivate testedMethod(response,
```
Comment (seglo) [inline s3/src/test/scala/akka/stream/alpakka/s3/impl/S3StreamSpec.scala:162]: TIL of ScalaTest's private method reflection API

File: s3/src/test/scala/akka/stream/alpakka/s3/impl/SplitAfterSizeSpec.scala [hunk 5]
```diff
@@ -29,8 +28,6 @@ class SplitAfterSizeSpec(_system: ActorSystem)
   def this() = this(ActorSystem("SplitAfterSizeSpec"))
   implicit val defaultPatience: PatienceConfig = PatienceConfig(1.seconds, 5.millis)
 
-  implicit val materializer = ActorMaterializer(ActorMaterializerSettings(system).withDebugLogging(true))
```
Comment (seglo) [inline s3/src/test/scala/akka/stream/alpakka/s3/impl/SplitAfterSizeSpec.scala:32]: I'm guessing this was a leftover from a troubleshooting session.

**Review by seglo — COMMENTED** [review]
Left some questions. Not sure when we should or shouldn't use the DEBUG level logging.

File: s3/src/main/scala/akka/stream/alpakka/s3/javadsl/S3.scala (hunk 1 above)
Comment (ennru) [reply s3/src/main/scala/akka/stream/alpakka/s3/javadsl/S3.scala:891]: Added deprecations.

File: s3/src/test/scala/akka/stream/alpakka/s3/impl/MemoryBufferSpec.scala (hunk 2 above)
Comment (ennru) [reply s3/src/test/scala/akka/stream/alpakka/s3/impl/MemoryBufferSpec.scala]: Removed all of those now.

File: s3/src/test/scala/akka/stream/alpakka/s3/impl/S3StreamSpec.scala (hunk 4 above)
Comment (ennru) [reply s3/src/test/scala/akka/stream/alpakka/s3/impl/S3StreamSpec.scala:162]: Should we use `.`-syntax instead of infix?

**Review by ennru — COMMENTED** [review]
Adapted a bit more to avoid warnings in Scala 2.13.

File: s3/src/test/scala/akka/stream/alpakka/s3/impl/S3StreamSpec.scala (hunk 4 above)
Comment (seglo) [reply s3/src/test/scala/akka/stream/alpakka/s3/impl/S3StreamSpec.scala:162]: I don't really have a preference here.

**Review by seglo — APPROVED** [review]
LGTM

File: s3/src/test/scala/akka/stream/alpakka/s3/impl/S3StreamSpec.scala (hunk 4 above)
Comment (raboof) [reply s3/src/test/scala/akka/stream/alpakka/s3/impl/S3StreamSpec.scala:162]: this is a bit obscure anyway, so either way works for me. Probably a *little* nicer to use `.`-syntax.

**Review by raboof — APPROVED** [review]

**Pull request merged**